test/CodeGenILTests/ReplicationTests.cs

```diff
@@ -724,5 +720,43 @@ public void Value_Are_Correct_Reversed()
             Assert.AreEqual(0.5, output["a"]);
             Assert.AreEqual(10.0, output["b"]);
         }
+
+        [Test]
+        public void IEnumerable_IntDouble_Coercion()
+        {
+            string code =
+            @"
+            import(""DesignScriptBuiltin.dll"");
+            import(""DSCoreNodes.dll"");
+            import (""FFITarget.dll"");
+            x = [1, 2, 3.7, 4];
+            test1 = FFITarget.DummyCollection.ReturnIEnumerableOfInt(x);  
+            test2 = DSCore.Math.Sum(test1);    
+            ";
+            var ast = ParserUtils.Parse(code).Body;
+            var output = codeGen.EmitAndExecute(ast);
+            Assert.IsNotEmpty(output);
+            Assert.AreEqual(new int[] { 1, 2, 4, 4 }, output["test1"]);
+            Assert.AreEqual(11, output["test2"]);
+        }
+
+        [Test]
+        public void IList_T_IEnumerable_Coerce()
```

Comment:
perhaps it's now worth adding a test that asserts that the generated IL does not include any call instructions to ReplicationLogic?